generic/tests/netperf.py

```diff
@@ -130,6 +130,13 @@ def mtu_set(mtu):
         if netdst in br_in_use:
             ifaces_in_use = host_bridges.list_iface()
             target_ifaces = list(ifaces_in_use + br_in_use)
+        if params.get("netdst_nic1") in process.system_output(
+                "ovs-vsctl list-br", ignore_status=True, shell=True).decode():
+            ovs_list = "ovs-vsctl list-ports %s" % params["netdst_nic1"]
+            ovs_port = process.system_output(ovs_list,
+                                             shell=True).decode().splitlines()
+            target_ifaces = target_ifaces + list([params.get("netdst_nic1")]
+                                                 + list(ovs_port))
```

Comment:
`ovs_port` is generated from `splitlines()`, so it is a list, do not need `list()` convert again. `target_ifaces` is originally a `set`. So you might want use `target_ifaces.update(ovs_port.append(params.get("netdst_nic1")))`

Reply:
it will be error "ERROR: 'list' object has no attribute 'update'"

Reply:
As you suggest, using ovs_port.append(params.get("netdst_nic1")), the error is "ERROR: can only concatenate list (not "NoneType"

Use "target_ifaces = target_ifaces + list(params.get("netdst_nic1")) + ovs_port", the target_ifaces is target_ifaces=['virbr0-nic', 'em1', 't1-e4HPoP', 't1-aLy5it', 't1-rzqYrO', 'virbr0', 'switch', 'o', 'v', 's', '0', 'p5p2', 't0-2zUNX8', 't0-6IsIti'], it will occurs avocado.utils.process.CmdError: Command 'ifconfig o mtu 9000' failed."

So, it seems original code is correct.

Reply:
> As you suggest, using ovs_port.append(params.get("netdst_nic1")), the error is "ERROR: can only concatenate list (not "NoneType"
>
> Use "target_ifaces = target_ifaces + list(params.get("netdst_nic1")) + ovs_port", the target_ifaces is target_ifaces=['virbr0-nic', 'em1', 't1-e4HPoP', 't1-aLy5it', 't1-rzqYrO', 'virbr0', 'switch', 'o', 'v', 's', '0', 'p5p2', 't0-2zUNX8', 't0-6IsIti'], it will occurs avocado.utils.process.CmdError: Command 'ifconfig o mtu 9000' failed."

Thanks for the information, seems we get more issues exposed. I am really curious about what you get for `ovs_port` and `params.get("netdst_nic1")`.
From the log, `NoneType` shows `netdst_nic1` is `None`, so we should have a judgement of it if so.
If you got `['o', 'v', 's', '0']` for `ovs_port`, the proper way to fix the issue should not be adding extra `list()`, you might try to get right string `ovs0` for it, in L#136.

Reply:
@quanwenli Okay, so I suppose `ovs_port` is None, then it should be:
```
target_ifaces.append(params["netdst_nic1"])
if ovs_port is not None:
    target_ifaces.extend(ovs_port)
```

Reply:
BTW, the issue 1-3 is caused by `params.get["netdst_nic1"]`, nothing to do with other usage. You can't use `[]` to express a method - `get()`.

Reply:
@quanwenli If you insist use `+`, then it should be `target_ifaces + [params["netdst_nic1"]] + ovs_port`

Reply:
I have updated, please help review. the results are all passed, please check in bzs.